content/post/twitteralgo.md

```diff
@@ -0,0 +1,186 @@
+
+
+-*What Tweets did the people I follow recently engage with?*\
+-*Who likes similar Tweets to me, and what else have they recently liked?*\
+-*What Tweets and Users are similar to my interests?*
+
+![image](/blog/images/twitteralgochart.png#center)
+
+
+## Multiplier System
+
+There is also a multiplier system where the reachability of tweets is impacted by certain parameters such as likes, replies, etc. Likes have a multiplier boost of 30 times and retweets have a boost of 20 times. An extensive list of these multiplier levels can be found in the code attatched below. This is a great way to get insight into how a tweet can ahieve better reachability and visibility. 
+
+It should be kept in mind though that offensive of NSFW tweets might not rank well, and even spamming multiple hashtags might get decrease the visibility of a tweet, but trending topics also might get you a small boost at the same time.
+
+
+
+```
+//Code found in the twitter open source algorithm
+
+private def getLinearRankingParams: ThriftRankingParams
+ThriftRankingParams(
+"type' = Some (ThriftScoringFunctionType.Linear),
+minScore = -1. 0e100,
+retweetCountParams
+= Some (ThriftLinearfeatureRankingParams (weight = 20.0))
+replyCountParams = Some (ThriftLinearFeatureRankingParams (weight = 1.0)),
+reputationParams = Some (ThriftLinearfeatureRankingParams (weight = 0.2)),
+luceneScoreParams = Some(ThriftLinearFeatureRankingParams (weight = 2.0))
+textScoreParams = Some(ThriftlinearFeatureRankingParams (weight = 0.18)),
+urlParams = Some(ThriftLinearFeatureRankingParams (weight = 2.0)),
+isReplyParams = Some (ThriftLinearFeatureRankingParams (weight = 1.0)),
+favCountParams = Some (ThriftLinearFeatureRankingParams (weight = 30.0)),
+langEnglishUlBoost = 0.5,
+langEnglishTweetBoost = 0.2,
+langDefaultBoost = 0.02,
+unknownLanguageBoost = 0.85,
+offensiveBoost = 0.1,
+inTrustedCircleBoost = 3.0,
+multipleHashtags0rTrendsBoost = 0.6,
+inDirectfollowBoost = 4.0,
+tweetHasTrendBoost = 1.1,
+selfTweetBoost = 2.0,
+tweetHasImageUrlBoost = 2.0,
+tweetHasVideoUrlBoost = 2.0,
+useUserLanguageInfo = true,
+ageDecayParams = Some (ThriftAgeDecayRankingParams (slope = 0.005, base = 1.0))
+
+```
+
+## Does Your Follower-Following Ratio Play a Role?
+
+As suggested in the code attatched below, if a user is following way more users than the number of people following them, their tweets may get a nerf in terms of ranking. The quality of users you interact with also plays a role, as interacting with bots, spam accounts or accounts with lower rankings might also get you a demotion. This entire system does play out almost like an ELO ranking system at times!
+
+```
+//reduce pagerank of users with low followers but high followings
+
+def adjustReputationsPostCalculation(mass: Double, numFollowers: Int, numFollowings: Int) = {
+if (numFollowings > threshAbsNumFriendsReps) {
```

Comment:
Shouldn't there be indentation here (and the rest of the function? not sure how it's rendered

Reply:
syntax highlighting is also messed up. apparently this theme doesn't support scala highlighting.

Reply:
I did struggle with it quite a bit. Do you think attaching IDE screenshots with proper highlighting would be a better idea?

Reply:
I don't see him specifying scala language, how would it be highlighted?

For reference, need `scala` after the 3 starting backticks:
![image](https://user-images.githubusercontent.com/72784348/233794157-4e5eca0c-092f-4a07-add4-3324971bcfc8.png)

Reply:
@ADV1K @RaghavS8Y approve #109

Reply:
Have approved #109, and added scala after backticks. Also changed the syntax in the first code block.

Reply:
@RaghavS8Y do a `git pull --rebase`. You should see the syntax highlighting work. Finally a `git push -f` to update the PR.

Reply:
Done, the highlighting seems to be working now, thanks so much!